kornia/core/image.py

```diff
@@ -0,0 +1,280 @@
+from enum import Enum
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from kornia.core import Tensor
+from kornia.utils import image_to_tensor, tensor_to_image
+
+
+class ImageColor(Enum):
+    r"""Enum that represents an image color space."""
+    GRAY = 0
+    RGB = 1
+    BGR = 2
+    NONE = 4
```

Comment:
I introduced that for testing -- we can remove. However, depending how clever we want this class, but we could arrive after some combinations to an undefined space. One example is when we pass this to a network or similar -- the resulting object will still be an image (unless we use `__torch_function__` to somehow modify) with also a resulting unkown number of channels in the feature space.

Reply:
One more consideration, is that the current implementation doesn't work well with torchscript because is not able to trace the custom class members like color, mean, or std.

An alternative to implement this API as a composition instead of inheritance which will easy later as planned to support other things than pytorch, or use directly our rust backend with `cv::Tensor`. Open to suggestion here.